Verso Books.js

```diff
@@ -0,0 +1,315 @@
+{
+	"translatorID": "bc517bd2-4835-4621-b775-917253bec93a",
+	"label": "Verso Books",
+	"creator": "Bo An",
+	"target": "^https?://(www\\.)?versobooks\\.com/books/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2021-06-29 07:01:44"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2021 Bo An
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	const isSubject = url.includes("subjects/");
+	if (isSubject && getSearchResults(doc, true)) {
+		return 'multiple';
+	}
+	else if (doc.querySelector('.edition-single--book-title')) {
+		return 'book';
+	} else {
+		return false;
+	}
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), (items) => {
+			if (!items) {
+				return true;
+			}
+			const articles = [];
+			for (const i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+			return true;
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+function getSearchResults(doc) {
+	let items = {};
+	let found = false;
+
+	const bookLinkEls = doc.querySelectorAll('.book-grid > .book-card');
+
+	Z.debug(bookLinkEls.length);
+
+	bookLinkEls.forEach((bookLinkEl) => {
+		const titleEl = bookLinkEl.querySelectorAll('div.book-card--title > a')[0];
+
+		const href = titleEl.href;
+		const title = titleEl.textContent;
+
+		items[href] = title;
+		if (found === false) {
+			found = true;
+		}
+	});
+
+	return found ? items : false;
+}
+
+function scrape(doc, _) {
+	const newItem = new Zotero.Item('book');
+
+	newItem.publisher = "Verso Books";
+
+	const title = doc.getElementsByClassName('edition-single--book-title')[0].textContent;
+	const subtitle = doc.getElementsByClassName('edition-single--book-subtitle')[0].textContent;
```

Comment:
When you're just pulling the `textContent` of a selector, it's best to use `text` so that the translator doesn't error out if the selector doesn't match. Like this:

```suggestion
	const title = text(doc, '.edition-single--book-title');
	const subtitle = text(doc, '.edition-single--book-subtitle');
```

Reply:
Got it! Thanks Abe!